### DISCUSSION TITLE
Dynamically remove and restore devices (after setup)

### DISCUSSION THREAD

**Discussion by jsve** [HomeSpan/HomeSpan#271] [Q&A]
Fist of all:
Great project! I was deep in the pit of trying to set this up with the raw C/C++ libraries a year ago, but you have made things a hole lot easier! The tutorials and documentation are all exceptional!

To my question:
I have a device with configuration-switches on it. Depending on the state of these I'd like to expose different services and devices to HomeKit. For simplicity, let's say that I have 4 LightBulbs connected, but depending on how the switches are configured I only want to expose 1, 2 or all 4 of the LightBulbs to the HomeKit ecosystem.

I guess what would solve this is a way to access all of the created SpanAccessories and remove/add them back in. This might be possible during setup, but like the title says I'd like to do it while the device is running (without restart). Ideal would be to be able to just disable devices, but from what I can tell, HomeKit does not support that.

I know this is a pretty extreme edge-case and might not be possible with the current architecture.

**Comment by HomeSpan** [comment]
This is a really good request.  At present you can dynamically create SpanAccessories, but after the first call to homeSpan.poll(), everything is fixed in place.  I have a sketch that allows you to make changes to the Accessory configuration via a web interface (see [Programmable Hub Example](https://github.com/HomeSpan/ProgrammableHub)), but a reboot is required to reset the device to the new config.

Changing the config on-the-fly should be possible but will require some modifications to HomeSpan.  Among other things, I will need to create destructors for SpanCharacteristic, SpanService, and SpanAccessory, and then a method to cause all open controller connections to drop so that are forced to re-read the new database.   I'm in the middle of version 1.5.1 but will look into this for version 1.5.2.
